app/connectors/ExternalPropertyLinkConnector.scala

```diff
@@ -0,0 +1,99 @@
+                                              @Named("voa.myClientsPropertyLink") myClientsPropertyLinkUrl: String,
+                                              @Named("voa.createPropertyLink") createPropertyLinkUrl: String,
+                                               conf: ServicesConfig) (implicit executionContext: ExecutionContext)
+  extends JsonHttpReads with OptionHttpReads with RawReads {
+
+
+
+
+  def getMyOrganisationsPropertyLinks(searchParams: GetPropertyLinksParameters, params: Option[PaginationParams])
+                                     (implicit hc: HeaderCarrier, request: ModernisedEnrichedRequest[_]): Future[Option[PropertyLinksWithAgents]] = {
+
+    println(myOrganisationsPropertyLinksUrl)
+    http.GET[Option[PropertyLinksWithAgents]](
+      myOrganisationsPropertyLinksUrl,
+      modernisedPaginationParams(params) ++
+        List(searchParams.address.map("address" -> _),
+          searchParams.baref.map("baref" -> _),
+          searchParams.agent.map("agent" -> _),
+          searchParams.status.map("status" -> _),
+          searchParams.sortfield.map("sortfield" -> _),
+          searchParams.sortorder.map("sortorder" -> _)).flatten)
+  }
+
+
+  def getMyOrganisationsPropertyLink(submissionId: String)
+                                    (implicit hc: HeaderCarrier, request: ModernisedEnrichedRequest[_]): Future[Option[OwnerPropertyLink]] =
+    http.GET[Option[OwnerPropertyLink]](myOrganisationsPropertyLinkUrl.replace("{propertyLinkId}", submissionId))
+
+  def getClientsPropertyLinks(searchParams: GetPropertyLinksParameters, params: Option[PaginationParams])
+                             (implicit hc: HeaderCarrier, request: ModernisedEnrichedRequest[_]): Future[Option[PropertyLinksWithClient]] =
+    http
+      .GET[Option[PropertyLinksWithClient]](
+      myClientsPropertyLinksUrl,
+      modernisedPaginationParams(params) ++
+        List(searchParams.address.map("address" -> _),
+          searchParams.baref.map("baref" -> _),
+          searchParams.agent.map("agent" -> _),
+          searchParams.status.map("status" -> _),
+          searchParams.sortfield.map("sortfield" -> _),
+          searchParams.sortorder.map("sortorder" -> _)).flatten)
+
+
+  def getClientsPropertyLink(submissionId: String)
+                            (implicit hc: HeaderCarrier, request: ModernisedEnrichedRequest[_]): Future[Option[ClientPropertyLink]] =
+    http.GET[Option[ClientPropertyLink]](myClientsPropertyLinkUrl.replace("{propertyLinkId}", submissionId))
+
+
+  def createPropertyLink(propertyLink: CreatePropertyLink)(implicit hc: HeaderCarrier, request: ModernisedEnrichedRequest[_]): Future[HttpResponse] =
+    http
+      .POST[CreatePropertyLink, HttpResponse](createPropertyLinkUrl, propertyLink, Seq())
+
+
+  private def modernisedPaginationParams(params: Option[PaginationParams]): Seq[(String, String)] =
+    params match {
```

Comment:
if you are doing this sort of statement why is the field optional?

Reply:
Haven't changed this yet, but will. Not sent as optional from frontend as far as I can see